include/arch/posix/arch.h

```diff
@@ -25,6 +25,7 @@
 #include <arch/posix/asm_inline.h>
 #include <board_irq.h> /* Each board must define this */
 #include <sw_isr_table.h>
+#include <posix_soc_if.h>
```

Comment:
This is a problem as long as posix_sic_if.h is a private header.

Reply:
It can "public", it is just not meant to be used directly by non POSIX arch/soc/boards files

Reply:
include/arch/posix/arch.h is a public header. It must not include any private headers. If we implement the changes in #19666 (i.e. remove arch/*/include/ from global include path), including this file outside kernel and arch will result in "posix_soc_if.h" cannot be found. Note that arch/cpu.h is included by kernel.h.

Reply:
For this PR, it should be okay to leave it as is. I will move the necessary externs (e.g. posix_irq_lock and posix_irq_lock_unlock) to include/arch/posix/arch.h in #20119.

Reply:
Yep I understand, but that situation exists without this PR too.
This PR cuts the dependency chain. So now having this header be included from other headers will not cause trouble anymore.